src/main/java/org/opensearch/security/privileges/dlsfls/AbstractRuleBasedPrivileges.java

```diff
@@ -149,7 +150,18 @@ public boolean isUnrestricted(PrivilegesEvaluationContext context, IndexResolver
         // If we found an unrestricted role, we continue with the next index/alias/data stream. If we found a restricted role, we abort
         // early and return true.
 
-        for (String index : resolved.getAllIndicesResolved(context.getClusterStateSupplier(), context.getIndexNameExpressionResolver())) {
+        Set<String> resolvedIndices = resolved.getAllIndicesResolved(
+            context.getClusterStateSupplier(),
+            context.getIndexNameExpressionResolver()
+        );
+        Set<String> matchedSystemIndices = SystemIndexRegistry.matchesSystemIndexPattern(resolvedIndices);
+
+        Set<String> regularIndices = new HashSet<>(resolvedIndices);
```

Comment:
This isn't quite right. The issue also manifests of the follower cluster contains indices that don't match the index pattern in the role definition. The logic in here needs to be looked into further to determine if this should be iterating over concrete indices for cluster actions that do not pertain to indices.

Reply:
Switched the logic to skip the FlsDlsValveImpl for any cluster perm.